### PR TITLE
Add tests for Config and ServiceProvider

### PR DESCRIPTION
**Pull request by hcjulz**
This PR adds table tests for creating new configs and cleans up a little bit of code.

### CODE REVIEW

File: saml/config.go [hunk 1]
```diff
@@ -1,105 +1,73 @@
...
-	SSOServiceURL string
-
-	// TODO: Add more fields for the cert, crypto algorithm, etc.
-	// SLOServiceURL string
-}
-
 type Config struct {
-	// ServiceBinding defines the mechanism that should be used
-	// when exchanging messages between the requester (SP) and responder (IDP).
-	ServiceBinding core.ServiceBinding
```
Comment (hcjulz) [inline saml/config.go:32]: We don't really need this here. The idea in the beginning was to configure what kind of binding should be used, but that can be decided when creating the SAML AuthnRequest.

File: saml/config.go [hunk 2]
```diff
@@ -1,105 +1,73 @@
...
-
-	// NameIDFormat controls how the users at the IDP are mapped to
-	// users at the SP during SSO.
-	NameIDFormat core.NameIDFormat
```
Comment (hcjulz) [inline saml/config.go:36]: I made this the default for the AuthnRequest and the plan is to add options to the create AuthnRequest methods later.

File: saml/config.go [hunk 3]
```diff
@@ -1,105 +1,73 @@
...
 	MetadataURL *url.URL
 
-	// ValidUntil is a function that defines until the generate service provider metadata
-	// document is valid.
+	// ValidUntil is a function that defines until the generated service provider metadata
+	// document is valid. (optional)
 	ValidUntil ValidUntilFunc
-
-	// Certificate is used to sign SAML Authentication Requests.
-	Certificate *tls.Certificate
```
Comment (hcjulz) [inline saml/config.go:56]: As we don't plan to sign Auth request in first place I removed it for now. We can add it back when implementing that feature.

File: saml/config.go [hunk 4]
```diff
@@ -1,105 +1,73 @@
...
-func NewConfig(entityID, acs, issuer, metadata *url.URL) *Config {
-	return &Config{
+func NewConfig(entityID, acs, issuer, metadata *url.URL) (*Config, error) {
+	const op = "saml.NewConfig"
+
+	cfg := &Config{
 		EntityID:                    entityID,
 		Issuer:                      issuer,
 		AssertionConsumerServiceURL: acs,
 		ValidUntil:                  DefaultValidUntil,
```
Comment (hcjulz) [inline saml/config.go:42]: I left this for now. I'd make this an option later too.

File: saml/config.go (hunk 3 above)
Comment (austingebauer) [reply saml/config.go:56]: Sounds good to me 👍

File: saml/config.go [hunk 5]
```diff
@@ -1,105 +1,73 @@
...
-	// when exchanging messages between the requester (SP) and responder (IDP).
-	ServiceBinding core.ServiceBinding
-
-	// NameIDFormat controls how the users at the IDP are mapped to
-	// users at the SP during SSO.
-	NameIDFormat core.NameIDFormat
-
 	// AssertionConsumerServiceURL defines the endpoint at the SP where the IDP
-	// will redirect to with its authentication response.
+	// will redirect to with its authentication response. (required)
```
Comment (austingebauer) [inline saml/config.go:18]: Nice addition of required/optional for the godocs.

**Comment by austingebauer** [comment]
A little late here. Thanks, @hcjulz. Looks good ✅

**Pull request merged**